node/src/Commands.ts

```diff
@@ -494,6 +494,300 @@ export function createSetBit(
+     */
+    constructor(offset: number) {
+        this.offset = offset.toString();
+    }
+
+    public toArg(): string {
+        return this.offset;
+    }
+}
+
+/**
+ * Represents an offset in an array of bits for the {@link BaseClient.bitfield|bitfield} or
+ * {@link BaseClient.bitfieldReadOnly|bitfieldReadOnly} commands. The bit offset index is calculated as the numerical
+ * value of the offset multiplied by the encoding value.
+ *
+ * For example, if we have the binary 01101001 with offset multiplier of 1 for an unsigned encoding of size 4, then the
+ * value is 9 from `0110(1001)`.
+ */
+export class BitOffsetMultiplier implements BitFieldOffset {
+    private static readonly OFFSET_MULTIPLIER_PREFIX = "#";
+    private readonly offset: string;
+
+    /**
+     * Creates an instance of BitOffsetMultiplier.
+     *
+     * @param offset - The offset in the array of bits, which will be multiplied by the encoding value to get the final
+     *      bit index offset.
+     */
+    constructor(offset: number) {
+        this.offset = `${BitOffsetMultiplier.OFFSET_MULTIPLIER_PREFIX}${offset.toString()}`;
+    }
+
+    public toArg(): string {
+        return this.offset;
+    }
+}
+
+/**
+ * Represents subcommands for the {@link BaseClient.bitfield|bitfield} or
+ * {@link BaseClient.bitfieldReadOnly|bitfieldReadOnly} commands.
+ */
+export interface BitFieldSubCommands {
+    /**
+     * Returns the subcommand as a list of string arguments to be used in the {@link BaseClient.bitfield|bitfield} or
+     * {@link BaseClient.bitfieldReadOnly|bitfieldReadOnly} commands.
+     *
+     * @returns The subcommand as a list of string arguments.
+     */
+    toArgs(): string[];
+}
+
+/**
+ * Represents the "GET" subcommand for getting a value in the binary representation of the string stored in `key`.
+ */
+export class BitFieldGet implements BitFieldSubCommands {
+    private static readonly GET_COMMAND_STRING = "GET";
+    private readonly encoding: BitEncoding;
+    private readonly offset: BitFieldOffset;
+
+    /**
+     * Creates an instance of BitFieldGet.
+     *
+     * @param encoding - The bit encoding for the subcommand.
+     * @param offset - The offset in the array of bits from which to get the value.
+     */
+    constructor(encoding: BitEncoding, offset: BitFieldOffset) {
+        this.encoding = encoding;
+        this.offset = offset;
+    }
+
+    toArgs(): string[] {
+        return [
+            BitFieldGet.GET_COMMAND_STRING,
+            this.encoding.toArg(),
+            this.offset.toArg(),
+        ];
+    }
+}
+
+/**
+ * Represents the "SET" subcommand for setting bits in the binary representation of the string stored in `key`.
+ */
+export class BitFieldSet implements BitFieldSubCommands {
+    private static readonly SET_COMMAND_STRING = "SET";
+    private readonly encoding: BitEncoding;
+    private readonly offset: BitFieldOffset;
+    private readonly value: number;
```

Comment:
Any reason for adding `readonly`? A user can save options to be passed around to different calls of the commands. With this change though we are restricting users.

Reply:
I figured users would probably create new objects instead of modifying existing ones, but I can change this if we think it would be better without the `readonly`